### PR TITLE
feat: add ApproximateReceiveCount to the message attributes

### PR DESCRIPTION
**Pull request by FahadAijaz**
1. Add the `ApproximateReceiveCount` option to the `attributeNames` for the `ConsumerOptions`.
2. Add the `approximateReceiveCount` from the `Attributes` to the `MessageMetadata`.

### CODE REVIEW

File: src/emitters/emitter.sqns.ts [hunk 1]
```diff
@@ -997,4 +999,19 @@ export class SqnsEmitter implements IEmitter {
     const urlParts = queueUrl.split('/')
     return urlParts[urlParts.length - 1]
   }
+
+  private saveApproximateReceiveCount(receivedMessage: Message, message: IMessage<any>) {
+    const approximateRecievedCount = this.getApproximateReceiveCount(receivedMessage);
+    if (message.messageAttributes) {
```
Comment (MGB247) [inline src/emitters/emitter.sqns.ts:1005]: if messageAttributes do not exist, we should create them?

File: src/emitters/emitter.sqns.ts [hunk 2]
```diff
@@ -997,4 +999,19 @@ export class SqnsEmitter implements IEmitter {
...
+      message.messageAttributes['ApproximateReceiveCount'] = {
```
Comment (MGB247) [inline src/emitters/emitter.sqns.ts:1006]: add this to MessageMetaData

Comment (FahadAijaz) [reply src/emitters/emitter.sqns.ts:1006]: yes Rayyan suggested, the same. Implementing this. Thanks

File: src/emitters/emitter.sqns.ts (hunk 1 above)
Comment (FahadAijaz) [reply src/emitters/emitter.sqns.ts]: Got it, done, please check.

File: src/emitters/emitter.sqns.ts (hunk 2 above)
Comment (FahadAijaz) [reply src/emitters/emitter.sqns.ts]: done, please check.

**Review by MGB247 — APPROVED** [review]

**Review by MGB247 — CHANGES_REQUESTED** [review]
Remember to up the version

**Review by MGB247 — APPROVED** [review]

**Pull request merged**